src/bases/base.js

```diff
@@ -92,8 +98,12 @@ class Decoder {
    */
   decode (text) {
     if (typeof text === 'string') {
-      switch (text[0]) {
-        case this.prefix: {
+      switch (text.codePointAt(0)) {
+        case this.prefixCodePoint: {
+          if (this.prefixCodePoint > 255) {
+            // special case for base256emoji, slow and hacky
+            return this.baseDecode(Array.from(text).slice(1).join(''))
```

Comment:
How about this instead ?

```suggestion
            return this.baseDecode(text.slice(this.prefixCodePoint.length))
```

Reply:
From what I understand this split the slices at regular intervals each 4 bytes (since :rocket: is 4 bytes big).

That would be too easy, codepoints are 2 to 6 bytes big (in utf8) else that would be no fun.

Reply:
@Jorropo I think maybe you're thinking `split()`, as `slice(x)` is only going to slice off the prefix of the string, which is what we want to do here (although we want `this.prefix.length` as the number of chars).

Good suggestion and it removes this special case. I'm converting this `switch` to a simpler `if` and I'm also going to base256emoji to `'multiformats/basics'` so that'll remove the other special case. Pushing that now.

Reply:
> I think we could simply this as by doing:
> 
> ```
> switch (text.codePointAt(0) {
>   case this.prefix:
>      return this.baseDecode(text.slice(this.prefix.length))
> // ...
> }
> ```

no, cause you're comparing a codePoint (number) to a prefix (string)